icicle_v3/backend/cpu/include/tasks_manager.h

```diff
@@ -0,0 +1,299 @@
+#pragma once 
+#include <atomic>
+#include <thread>
+#include <stdexcept>
+#include <cassert>
+
+#define LOG_TASKS_PER_THREAD 2
```

Comment:
Shouldn't we set that based on cpu arch? for future optimization

Reply:
it's not the number of threads but the size of the threads' fifo. It's currently set heuristically to reduce the idle time of worker threads.